### PR TITLE
Nick dev zenodo

### LINKED ISSUE #3: Automatic Zenodo Backup
Need to figure out how to automate the Zenodo backup of data files and iffy files created each month by this project.

- [x] Nick was going to try and pick this up from what Matt had started but I cannot add him to the assignees list because he is not an organization member. Let's all try to remember to add him to this issue once he is added to the organization.

#### Helpful information:
- See: https://developers.zenodo.org/?python#quickstart-upload
- Matt created dedicated Top FIBers API keys for this project
    - Nick/Pasan have access to Zenodo API keys in the Top FIBers slack channel
    - These can also be accessed (or updated/refreshed) on Zenodo. See the [Truthy Accounts page](https://github.iu.edu/truthy-team/truthy/wiki/Truthy-Team-Accounts) for login information to Zenodo and to update the API key
- We have already created records that exist within the OSoMe Zenodo Community
    - Forever Zenodo (cannot be deleted): https://zenodo.org/record/7905198
    - Sandback Zenodo (for testing!): https://sandbox.zenodo.org/record/1199606
- You can see the test code that Matt began to develop but did not finish in the  `zenodo_backup` branch of the old repo that Matt created (see [this](https://github.com/mr-devs/top-FIBers/tree/zenodo_backup/scripts/data_backup))

### LINKED ISSUE #12: Add git line for iffy data files to monthly bash script
We need to add a line to the monthly pipeline that backs up the latest iffy domains file to GitHub. I think the below will work but we will need to do some testing.

```
git add data/iffy_files/*; git commit -m 'add latest iffy file; git push
```

Not sure if this is better to add directly after the iffy list has been downloaded or at the end of the pipeline... 🤔 

> **NOTE:** We may want to make the above more specific. Right now it will add anything that is in there, which is a bit of overkill. We could instead be more specific about the file name. However, this would require getting the date right which would take some more effort.

### PR DESCRIPTION
**Pull request by nick-c-liu**
Here are some changes I made, which should take care of the zenodo backup and iffy updates. Zenodo worked on Jul 1<sup>st</sup> 's data, but we shall see how the big script runs next month. 
- added scripts for zenodo updates
- added git operations for updating iffy files

Thank you guys for reviewing the changes, please make comments on where things doesn't look right.

---
Closes #12 
Closes #3

### CODE REVIEW

File: scripts/update_zenodo.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            logger.info("No updates detected")
+            rm_draft(ACCESS_TOKEN, new_deposition_id)
+        else:
+            # publish draft
+            url_publish = f"https://zenodo.org/api/deposit/depositions/{new_deposition_id}/actions/publish"
+            response = requests.post(url_publish, params={'access_token': ACCESS_TOKEN})
+
+            if response.status_code == 202:
+                logger.info(f"New version published. ID: {new_deposition_id}")
+            else:
```
Comment (mr-devs) [inline scripts/update_zenodo.py:152]: I think we need to create the `success.log` file here after the new version is published successfully.

File: scripts/update_zenodo.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        logger.info(f"New deposition ID: {new_deposition_id}")
+
+        # Retrieve list of local files
+        file_names = [f for f in os.listdir(folder_path) if os.path.isfile(os.path.join(folder_path, f)) and not f.startswith('.')]
+        remote_files = {x['filename']:(x['checksum'],x['id']) for x in json_response["files"]}
+        to_be_updated = []
+        # iterate through local files and either update or upload
+        for filename in file_names:
+            file_path = f"{folder_path}/{filename}"
+            if filename in remote_files:
```
Comment (mr-devs) [inline scripts/update_zenodo.py]: Why do we need this block? Old files should not be changed/updated, I don't think. Is that right? I am worried that something weird will happen and we'll accidentally delete all of the old files... Can we just skip files that we already have in the remote set? LMK what you think.

File: scripts/monthly_master_script.sh [hunk 3]
```diff
@@ -244,4 +244,24 @@ fi
...
+   echo "$(date -Is) : FAILED. Exiting <${SCRIPT_NAME}>." >> $MASTER_LOG
+   exit 1
+fi
+# Remove after checking for successful completion
+rm success.log
+
+
 echo "$(date -Is) : Script complete." >> $MASTER_LOG
+
+git add data/iffy_files
```
Comment (mr-devs) [inline scripts/monthly_master_script.sh]: This is a nice add. However, can you please log this in the same way as the other commands? You can even write the git line outputs directly to the master log.

e.g. ...
```bash
echo "$(date -Is) : Updating..." >> $MASTER_LOG
git add data/iffy_files >> $MASTER_LOG
git commit -m "adding new iffy files" >> $MASTER_LOG
git push >> $MASTER_LOG
```

As far as I know, this should work.

Then you can move the `echo "$(date -Is) : Script complete." >> $MASTER_LOG` line after this block. Thanks!

**Review by mr-devs — CHANGES_REQUESTED** [review]
This looks great! I have a couple of small requests/changes/questions. Let me know what you think... Thanks!!

File: scripts/update_zenodo.py (hunk 2 above)
Comment (nick-c-liu) [reply scripts/update_zenodo.py]: You are right, in our case the old files shouldn't change by any means. I'll skip these files instead.

File: scripts/update_zenodo.py (hunk 1 above)
Comment (nick-c-liu) [reply scripts/update_zenodo.py:152]: somehow missed that, added `success.log` at line 149

File: scripts/monthly_master_script.sh (hunk 3 above)
Comment (nick-c-liu) [reply scripts/monthly_master_script.sh]: block added👍🏻 Thanks!

**Review by nick-c-liu — COMMENTED** [review]
Thanks for spotting these! I've updated files accordingly, please let me know if something still need a fix. Thanks!

**Comment by mr-devs** [comment]
Looks great, thanks!

**Comment by mr-devs** [comment]
btw, i just pulled this down to the main branch on lenny. Should be good to go!!

**Comment by nick-c-liu** [comment]
Thanks for doing that!

**Pull request merged**